library/L1_Peripheral/msp432p401r/pin.hpp

```diff
@@ -0,0 +1,188 @@
+  /// code. Where the most significant bit determines the pin direction.
+  ///
+  /// @see Port Function Tables in 6.12 Input/Output Diagrams
+  ///      http://www.ti.com/lit/ds/symlink/msp432p401r.pdf#page=138
+  ///
+  /// @param function The 3-bit function code.
+  void SetPinFunction(uint8_t function) const override
+  {
+    constexpr auto kDirectionBit = bit::CreateMaskFromRange(2);
+    constexpr auto kSel1Bit      = bit::CreateMaskFromRange(1);
+    constexpr auto kSel0Bit      = bit::CreateMaskFromRange(0);
+
+    volatile uint8_t * select1_register = RegisterAddress(&Port()->SEL1);
+    volatile uint8_t * select0_register = RegisterAddress(&Port()->SEL0);
+    *select1_register = static_cast<uint8_t>(
+      bit::Insert(static_cast<uint32_t>(*select1_register),
+                  bit::Read(function, kSel1Bit),
+                  pin_));
+    *select0_register = static_cast<uint8_t>(
+      bit::Insert(static_cast<uint32_t>(*select0_register),
+                  bit::Read(function, kSel0Bit),
+                  pin_));
+
+    SetPinDirection(Gpio::Direction(bit::Read(function, kDirectionBit)));
+  }
+
+  /// Sets the pin's resistor pull either as pull up, pull down, or none.
+  ///
+  /// @see 12.2.4 Pullup or Pulldown Resistor Enable Registers (PxREN)
+  ///      https://www.ti.com/lit/ug/slau356i/slau356i.pdf#page=678
+  ///
+  /// @param resistor The resistor to set.
+  void SetPull(Resistor resistor) const override
+  {
+    volatile uint8_t * resistor_enable = RegisterAddress(&Port()->REN);
+    // The output register (OUT) is used to select the pull down or pull up
+    // resistor when resistor is enabled.
+    volatile uint8_t * resistor_select = RegisterAddress(&Port()->OUT);
+
+    switch (resistor)
+    {
+      case Resistor::kNone:
+        *resistor_enable = bit::Clear(*resistor_enable, pin_);
+        break;
+      case Resistor::kPullDown:
+        *resistor_enable = bit::Set(*resistor_enable, pin_);
+        *resistor_select = bit::Clear(*resistor_select, pin_);
+        break;
+      case Resistor::kPullUp:
+        *resistor_enable = bit::Set(*resistor_enable, pin_);
+        *resistor_select = bit::Set(*resistor_select, pin_);
+        break;
+      default: return;
+    }
+  }
+
+  /// Sets the drive strength of the pin.
+  ///
+  /// @param drive_strength The drive strength to set.
+  void SetDriveStrength(DriveStrength drive_strength)
+  {
+    volatile uint8_t * drive_strength_register = RegisterAddress(&Port()->DS);
+
+    if (drive_strength == DriveStrength::kHigh)
+    {
+      *drive_strength_register = bit::Set(*drive_strength_register, pin_);
+    }
+    else
+    {
+      *drive_strength_register = bit::Clear(*drive_strength_register, pin_);
+    }
+  }
+
+  void SetAsOpenDrain(bool) const override
+  {
+    // TODO(#1134): Should use Error-v2.
+    SJ2_ASSERT_FATAL(false, "This function is not implemented");
+  }
+
+  void SetAsAnalogMode(bool) const override {}
```

Comment:
Add a comment here stating that SetAnalogMode is not required here in order to enable ADCs.